public/compute/the_hairball.md

```diff
@@ -1,10 +1,174 @@
+
+|project |value|
+|--------|-----|
+|project1|1..2 |
+|project2|1..3 |
+
+They will be expanded to
+
+|project |value|
+|--------|-----|
+|project1|1    |
+|project1|2    |
+|project2|1    |
+|project2|2    |
+|project2|3    |
+
+Similarly, a parameter value with a comma-separated list of values will result in multiple rows, one for all values in the list. Surround the value with quotes to escape commas, e.g. `"sample2,sample3"`.
+
+So 
+
+|project |dir |sample         |
+|--------|----|---------------|
+|project1|dir1|sample1        |
+|project1|dir2|sample2,sample3|
+|project2|dir2|sample4        |
+
+Will be expanded to
+
+|project |dir |sample |
+|--------|----|-------|
+|project1|dir1|sample1|
+|project1|dir2|sample2|
+|project1|dir2|sample3|
+|project2|dir2|sample4|
+
 #### Multiple files
-#### i..j
-#### Collapsing (a.k.a. Folding)
+You may specify multiple parameter files.
+Each parameter may only be specified in one file.
+The resulting list of parameters will be created by taking all possible combinations of parameter combinations from each file.
+
+#### Simple example
+Say we have the following parameter combinations:
+
+|input|
+|-----|
+|hello|
+|bye  |
+
+and
+
+|workflowName   |creationDate|
+|---------------|------------|
+|myFirstWorkflow|today       |
+
+Then these will get combined to 
+
+|input|workflowName   |creationDate|
+|-----|---------------|------------|
+|hello|myFirstWorkflow|today       |
+|bye  |myFirstWorkflow|today       |
+
+#### Example with three files
+
+Say we have the following three files:
+
+|input|
+|-----|
+|hello|
+|bye  |
+
+|sample |
+|-------|
+|sample1|
+|sample2|
+
+|workflowName   |creationDate|
+|---------------|------------|
+|myFirstWorkflow|today       |
+
+Then these will get combined to 
+
+|input|sample |workflowName   |creationDate|
+|-----|-------|---------------|------------|
+|hello|sample1|myFirstWorkflow|today       |
+|hello|sample2|myFirstWorkflow|today       |
+|bye  |sample1|myFirstWorkflow|today       |
+|bye  |sample2|myFirstWorkflow|today       |
+
+#### Collapsing (or Folding) the table of Parameter values
+Say we have the following combinations:
+
+|project |dir |sample |
+|--------|----|-------|
+|project1|dir1|sample1|
+|project1|dir2|sample2|
+|project1|dir2|sample3|
+|project2|dir2|sample4|
+
+Now if a particular step's protocol depends only on the value of parameter `project`, it only needs to be run twice, once for `project=project1` and once for `project=project2`.
+So the table of parameter combinations can be collapsed to the much smaller table.
+
+|project |
+|--------|
+|project1|
+|project2|
+
+If the step's protocol depends on parameters `project` and `dir`, three instances for the step need to be created, collapsing the table as follows:
+
+|project |dir |
+|--------|----|
+|project1|dir1|
+|project1|dir2|
+|project2|dir2|
+
+The step need not be run for the combination `project=project1, dir=dir2`, since no parameter combinations exits with those values.
```

Comment:
The second  line is project 1, dir2 is it not? I think you meant project2, dir 1. That is the only combination that does not exist. "combinations exists with those"